app/services/proofing/resolution/progressive_proofer.rb

```diff
@@ -0,0 +1,190 @@
+module Proofing
+  module Resolution
+    # Uses a combination of LexisNexis InstantVerify and AAMVA checks to verify that
+    # a user's identity can be resolved against authoritative sources. This includes logic for when:
+    #   1. The user is or is not within an AAMVA-participating jurisdiction
+    #   2. The user has only provided one address for their residential and identity document
+    #      address or separate residential and identity document addresses
+    class ProgressiveProofer
+      def proof(
+        applicant_pii:,
+        double_address_verification:,
+        request_ip:,
+        should_proof_state_id:,
+        threatmetrix_session_id:,
+        timer:,
+        user_email:
+      )
+        device_profiling_result = proof_with_threatmetrix_if_needed(
+          applicant_pii: applicant_pii,
+          request_ip: request_ip,
+          threatmetrix_session_id: threatmetrix_session_id,
+          timer: timer,
+          user_email: user_email,
+        )
+
+        # todo(LG-8693): Begin verifing both the user's residential address and identity document
+        # address
+        applicant_pii = with_state_id_address(applicant_pii) if double_address_verification
+
+        resolution_result = proof_resolution(
+          applicant_pii: applicant_pii,
+          timer: timer,
+        )
```

Comment:
A future PR/ticket will update this method to also resolve the user's residential address if double_address_verification is true and adjudicate the combined result